### PR TITLE
Game stats

### PR DESCRIPTION
**Pull request by odellmac4**
Add Game class with following methods:
- highest_total_score
- lowest_total_score
- percentage_home_wins
- percentage_visitor_wins
- percentage_ties
- count_of_games_by_season
- average_goals_per_game
- average_goals_by_season
- calculate_percentage (helper method)

Additional Notes:
-changed name of instance variable in initialize method from data_game to data_games to make it plural

### CODE REVIEW

File: lib/stat_tracker.rb [hunk 1]
```diff
@@ -59,4 +59,73 @@ def read_game_teams_csv(location_path)
...
+         total_goals += (game.away_goals + game.home_goals)
+      end
+      (total_goals.to_f / @data_games.count).round(2)
+   end
+
+   def average_goals_by_season
+      goals_by_season = {}
+      @data_games.each do |game|
+         total_goals = game.away_goals + game.home_goals
+         if goals_by_season.key?(game.season)
```
Comment (odellmac4) [inline lib/stat_tracker.rb:114]: @IgorrMagalhaess could we refactor by just assigning the key for goals_by_season hash to game.season and the value would be total goals? (goals_by_season[game.season] = total_goals) Just a bit confused on why the if/else statement is needed here.

File: lib/stat_tracker.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def self.from_csv(location_path)
    end
 
    def initialize(locations_path)
-      @data_game = read_games_csv(locations_path[:games])
+      @data_games = read_games_csv(locations_path[:games])
```
Comment (IgorrMagalhaess) [inline lib/stat_tracker.rb:14]: This is great! it is an array so it's best practice to name variable in plural! great job!

File: lib/stat_tracker.rb (hunk 1 above)
Comment (IgorrMagalhaess) [reply lib/stat_tracker.rb:114]: We can but if we do that we have to do memoization before creating a new key if that key is not found. I can explain this better in our stand_up meeting today!
in this case would be:
`goals_by_season[game.season] ||= 0`
`goas_by_season[game.season] += total_goals`

Or we can initialize our hash in the variable `goals_by_season` with a value of '0'.

File: lib/stat_tracker.rb [hunk 3]
```diff
@@ -59,4 +59,73 @@ def read_game_teams_csv(location_path)
...
+      end
+      avg_goals_by_season = {}
+      goals_by_season.each do |season , total_goals|
+         avg_goals_by_season[season] = (total_goals.to_f / count_of_games_by_season[season]).round(2)
+      end
+      avg_goals_by_season
+   end
+
+#Helper Method
+   def calculate_percentage(num1 , num2)
```
Comment (IgorrMagalhaess) [inline lib/stat_tracker.rb:128]: I really like this helper method!

**Review by IgorrMagalhaess — APPROVED** [review]

**Pull request merged**